src/renderer/qt/qt_window_manager.cc

```diff
@@ -339,13 +339,67 @@ void FillCandidates(const commands::Candidates &candidates,
   const int width = kColumn0Width + max_width1 + max_width2 + kColumn3Width;
   table->resize(width, total_height);
 }
+
+class VirtualRect {
+ public:
+  static VirtualRect FromNativeRect(const Rect &native_rect) {
+    for (const QScreen *screen : QGuiApplication::screens()) {
+      Rect rect = TranslateToVirtual(screen, native_rect);
+      const QRect screen_rect = screen->geometry();
+
+      // Use top left to locate a screen
+      if (screen_rect.contains(rect.Left(), rect.Top())) {
+        return VirtualRect(rect, mozc::renderer::GetRect(screen->geometry()));
+      }
+    }
+
+    // fall back to primary screen
+    // TODO: Return the nearest monitor rect instead.
+    // See GetMonitorRect
+    const QScreen *screen = QGuiApplication::primaryScreen();
+    Rect rect = TranslateToVirtual(screen, native_rect);
+    return VirtualRect(rect, mozc::renderer::GetRect(screen->geometry()));
+  }
+
+  Rect GetRect() const { return rect_; }
+
+  Rect GetMonitorRect() const { return monitor_rect_; }
+
+ private:
+  VirtualRect(Rect rect, Rect monitor_rect)
```

Comment:
Does `const Rect &` work?

```
VirtualRect(const Rect &rect, const Rect &monitor_rect)

Reply:
Yes, it works. Changed them to const Rect &